cmds/tar/tar.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Create and extract tar archives.
+//
+// Synopsis:
+//     tar [-cxvf] DIRECTORY
+//
+// Description:
+//     There are only two ways to use this command line:
+//        tar -cvf x.tar directory/  # create
+//        tar -xvf x.tar directory/  # extract
+//
+// Options:
+//     -c: create a new tar archive from the given directory
+//     -x: extract a tar archive to the given directory
+//     -v: verbose, print each filename (optional)
+//     -f: tar filename (required)
+//
+// TODO: The arguments deviates slightly from gnu tar.
+package main
+
+import (
+	"log"
+	"os"
+
+	flag "github.com/spf13/pflag"
+	"github.com/u-root/u-root/pkg/tar"
+)
+
+var (
+	create  = flag.BoolP("create", "c", false, "create a new tar archive from the given directory")
+	extract = flag.BoolP("extract", "x", false, "extract a tar archive from the given directory")
+	verbose = flag.BoolP("verbose", "v", false, "print each filename")
+	file    = flag.StringP("file", "f", "", "tar file")
+)
+
+func main() {
+	flag.Parse()
+	if flag.NArg() != 1 {
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	if !*create == !*extract {
+		log.Fatal("either supply -c or -x")
+	}
+
+	if *file == "" {
```

Comment:
could you just use stdin/stdout if no file supplied?

Reply:
See #1168